src/components/Table.jsx

```diff
@@ -0,0 +1,112 @@
+import React, {Component, PropTypes} from 'react';
+import {connect} from 'react-redux';
+import {Paper, RefreshIndicator} from 'material-ui';
+import TableHeaderComponent from './TableHeaderComponent.jsx';
+import TableRowComponent from './TableRowComponent.jsx';
+import ReactPaginate from 'react-paginate';
+
+import Dialog from '../dialog/Dialog';
+import {
+  Table,
+  TableBody,
+  RaisedButton
+} from 'material-ui';
+
+const detailStyle = {
+  padding: 15,
+  margin: 15
+};
+const loadingIndicatorStyle = {
+  margin: 'auto',
+  position: 'relative'
+};
+
+class TableComponent extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      openModal: false,
+      actionModal: 'create'
+    };
+  }
+
+  handleOpenModal = () => {
+    this.setState({openModal: true, actionModal: 'create'});
+  };
+
+  handleCloseModal = () => {
+    this.setState({openModal: false});
+  };
+
+  handleSubmit = data => {
+    this.props.createData(this.props.schema.url, data, this.props.schema.singular);
+  };
+  handlePageClick() {
+
+  }
+  render() {
+    const {schema, singular} = this.props.schema;
+
+    if (this.props.isLoading || !Array.isArray(this.props.data)) {
+      return (
+        <RefreshIndicator size={60} left={0}
+          top={0} status="loading"
+          style={loadingIndicatorStyle}
+        />
+      );
+    }
+    return (
+      <Paper style={detailStyle}>
+        <RaisedButton onClick={this.handleOpenModal}
+          label={'Add new ' + singular}
+          style={{margin: '8px 8px 8px 8px', marginRight: 'auto'}}
+          primary={true}
+        />
+        <Dialog open={this.state.openModal} action={this.state.actionModal}
+          onRequestClose={this.handleCloseModal} schema={this.props.schema}
+          onSubmit={this.handleSubmit}
+        />
+        <Table style={{tableLayout: 'auto'}} className="table">
+          <TableBody displayRowCheckbox={false} showRowHover={true}>
+            <TableHeaderComponent schema={schema}/>
```

Comment:
I'm having trouble aligning header columns with body columns when it's moved to TableHeader